Orange/widgets/data/owtranspose.py

```diff
@@ -76,25 +106,40 @@ def update_controls(self):
             if self.feature_model:
                 self.feature_names_column = self.feature_model[0]
         enabled = bool(self.feature_model)
-        self.feature_radio.buttons[1].setEnabled(enabled)
+        self.feature_radio.buttons[self.FROM_META_ATTR].setEnabled(enabled)
         self.feature_combo.setEnabled(enabled)
-        self.feature_type = int(enabled)
+        self.feature_type = self.FROM_META_ATTR if enabled else self.GENERIC
+
+    def _apply_editing(self):
+        self.feature_type = self.GENERIC
+        self.custom_feature_name.setText(self.custom_feature_name.text().strip())
+        self.feature_name = self.custom_feature_name.text()
+        if self.auto_apply:
```

Comment:
If you added `if` to stop the changes from being applied when auto apply was disabled: the problem is that the auto apply button is a default button, hence Enter triggers it. To prevent this, use `self.apply_button.button.setAutoDefault(False)`.

Reply:
Exactly, that was the reason I added that `if`.